lib/pavilion/series.py

```diff
@@ -36,12 +49,236 @@ def test_obj_from_id(pav_cfg, test_ids):
+    """Series Manger"""
+
+    def __init__(self, pav_cfg, series_obj, series_cfg):
+        # set everything up
+
+        self.pav_cfg = pav_cfg
+        self.series_obj = series_obj
+        self.series_cfg = series_cfg
+
+        self.series_section = self.series_cfg['series']
+
+        self.dep_graph = {}  # { test_name: [tests it depends on] }
+        self.make_dep_graph()
+
+        universal_modes = self.series_cfg['modes']
+
+        self.test_info = {}
+
+        for test_name, test_config in self.series_section.items():
+            self.test_info[test_name] = {}
+            test_modes = test_config['modes']
+            all_modes = universal_modes + test_modes
+            self.test_info[test_name]['modes'] = all_modes
+            self.test_info[test_name]['only_if'] = test_config['only_if']
+            self.test_info[test_name]['not_if'] = test_config['not_if']
+
+        # create doubly linked graph
+        for test_name in self.dep_graph:
+            prev_list = self.dep_graph[test_name]
+            self.test_info[test_name]['prev'] = prev_list
+
+            next_list = []
+            for t_n in self.dep_graph:
+                if test_name in self.dep_graph[t_n]:
+                    next_list.append(t_n)
+            self.test_info[test_name]['next'] = next_list
+
+        # run tests in order
+        self.all_tests = list(self.test_info.keys())
+        self.started = []
+        self.finished = []
+        self.not_started = []
+        # kick off tests that aren't waiting on any tests to complete
+        for test_name in self.test_info:
+            if not self.test_info[test_name]['prev']:
+                self.run_test(test_name)
+                self.not_started = list(set(self.all_tests) - set(self.started))
+
+        while len(self.not_started) != 0:
+
+            self.check_and_update()
+            temp_waiting = copy.deepcopy(self.not_started)
+            for test_name in temp_waiting:
+                ready = all(wait in self.finished for wait in self.test_info[
+                    test_name]['prev'])
+                if ready:
+                    if self.series_section[test_name]['depends_pass'] == 'True':
+                        if self.all_tests_passed(
+                                self.test_info[test_name]['prev']):
+                            self.run_test(test_name)
+                            self.not_started.remove(test_name)
+                        else:
+                            temp_resolver = resolver.TestConfigResolver(
+                                self.pav_cfg)
+                            raw_configs = temp_resolver.load_raw_configs(
+                                [test_name], [],
+                                self.test_info[test_name]['modes']
+                            )
+                            for config in raw_configs:
+                                # need to delete these so they won't get
+                                # evaluated
```

Comment:
Change comment to:

`# Delete the conditionals - we're already skipping this test but for different reasons.`